src/models/Level.php

```diff
@@ -654,106 +654,156 @@ private static function levelFromRow(Map<string, string> $row): Level {
     );
   }
 
+  private static async function withLock(
+    string $lock_name,
+    int $team_id,
+    (function(): Awaitable<mixed>) $thunk,
+  ): Awaitable<mixed> {
+    $lock_name = sprintf('%s/%s_%d', sys_get_temp_dir(), $lock_name, $team_id);
```

Comment:
I was thinking of having each team reuse their lock file for the duration of the CTF. The benefit of this is we don't need to create a new file on each request. I don't see a downside of this reuse.